[Bug] `tpush_tpop_frontend_lowering_a3.pto` a3 codegen: TPipe argument type incorrect

### Component

EmitC / Codegen (lib/PTO/Transforms/PTOToEmitC.cpp)

### Description

`ptoas --pto-arch a3 PTOAS/test/basic/tpush_tpop_frontend_lowering_a3.pto` 

The input params to `TPipe` constructor are wrong
```
AICORE void cube_kernel(int32_t v1) {
  int32_t v2 = 512;
  int32_t v3 = 0;
  int64_t v4 = 0;
  using T = float;

  #if defined(__DAV_CUBE__)
  auto v5 = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(v1, v2, v3);
  Tile<TileType::Acc, float, 16, 16, BLayout::ColMajor, 16, 16, SLayout::RowMajor, 1024, PadValue::Null> v6;
  TASSIGN(v6, v4);
  TPUSH<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, Tile<TileType::Acc, float, 16, 16, BLayout::ColMajor, 16, 16, SLayout::RowMajor, 1024, PadValue::Null>, TileSplitAxis::TILE_NO_SPLIT>(v5, v6);
  Tile<TileType::Mat, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::NoneBox, 1024, PadValue::Null> v7;
  TPOP<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, Tile<TileType::Mat, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::NoneBox, 1024, PadValue::Null>, TileSplitAxis::TILE_UP_DOWN>(v5, v7);
  TFREE<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, TileSplitAxis::TILE_LEFT_RIGHT>(v5);
  #endif // __DAV_CUBE__

  return;
}
```

### Reproduction (minimal)

```markdown
1) Codegen

ptoas --pto-arch a3 test/basic/tpush_tpop_frontend_lowering_a3.pto > cross_comm.cpp

2) Compilation

bisheng \
    -I/sources/pto-isa/include/ \
    -fPIC -shared -D_FORTIFY_SOURCE=2 -O2 -std=c++17 \
    -Wno-macro-redefined -Wno-ignored-attributes -fstack-protector-strong \
    -xcce -Xhost-start -Xhost-end \
    --npu-arch=dav-2201 -DMEMORY_BASE \
    -std=gnu++17 \
    "cross_comm.cpp \
    -o cross_comm.so
```

### Expected behavior

`TPipe` arguments should have correct type.

### Actual behavior / error logs

```text
/mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/build_artifacts/bidirectional_example.cpp:30:13: error: no matching constructor for initialization of 'TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>'
  auto v5 = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(v1, v2, v3);
            ^                                         ~~~~~~~~~~
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:341:27: note: candidate constructor not viable: no known conversion from 'int32_t' (aka 'int') to '__gm__ void *' for 1st argument
    PTO_INTERNAL explicit TPipe(__gm__ void *GM_SLOT_BUFFER, uint32_t C2V_CONSUMER_BUF, uint32_t V2C_CONSUMER_BUF)
                          ^
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:30:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
struct TPipe {
       ^
In file included from /mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/caller.cpp:7:
/mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/build_artifacts/bidirectional_example.cpp:72:15: error: no matching constructor for initialization of 'TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>'
    auto v6 = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(v1, v5, v4);
              ^                                         ~~~~~~~~~~
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:341:27: note: candidate constructor not viable: no known conversion from 'int32_t' (aka 'int') to '__gm__ void *' for 1st argument
    PTO_INTERNAL explicit TPipe(__gm__ void *GM_SLOT_BUFFER, uint32_t C2V_CONSUMER_BUF, uint32_t V2C_CONSUMER_BUF)
                          ^
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:30:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
struct TPipe {
       ^
/mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/caller.cpp:14:5: error: use of undeclared identifier 'bidirectional_example'
    bidirectional_example<<<blockDim, nullptr, stream>>>((__gm__ float *)gmSlotBuffer);
    ^
3 errors generated.
In file included from /mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/caller.cpp:7:
/mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/build_artifacts/bidirectional_example.cpp:51:13: error: no matching constructor for initialization of 'TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>'
  auto v5 = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(v1, v3, v2);
            ^                                         ~~~~~~~~~~
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:341:27: note: candidate constructor not viable: no known conversion from 'int32_t' (aka 'int') to '__gm__ void *' for 1st argument
    PTO_INTERNAL explicit TPipe(__gm__ void *GM_SLOT_BUFFER, uint32_t C2V_CONSUMER_BUF, uint32_t V2C_CONSUMER_BUF)
                          ^
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:30:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
struct TPipe {
       ^
In file included from /mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/caller.cpp:7:
/mounted_home/pto-dsl/examples/aot/tpushpop/mix-kernel_mlir/build_artifacts/bidirectional_example.cpp:96:15: error: no matching constructor for initialization of 'TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>'
    auto v6 = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(v1, v4, v5);
              ^                                         ~~~~~~~~~~
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:341:27: note: candidate constructor not viable: no known conversion from 'int32_t' (aka 'int') to '__gm__ void *' for 1st argument
    PTO_INTERNAL explicit TPipe(__gm__ void *GM_SLOT_BUFFER, uint32_t C2V_CONSUMER_BUF, uint32_t V2C_CONSUMER_BUF)
                          ^
/sources/pto-isa/include/pto/npu/a2a3/TPush.hpp:30:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
struct TPipe {
```

### Git commit

1f0a12e8304b458020d0e4dc01ad62c75e80be98

### Host platform

Linux (aarch64)

### Target Ascend arch (if relevant)

a3

### PTOAS build level (if relevant)

None

--- a/docs/PTO_IR_manual.md
+++ b/docs/PTO_IR_manual.md
@@ -6811,6 +6811,13 @@ function's reserved buffer declaration.
 **Syntax:**
 
 ```mlir
+// A2/A3 (with GM slot buffer):
+pto.aic_initialize_pipe {dir_mask = 1, slot_size = 1024}
+  (gm_slot_buffer = %gm_buf : !pto.ptr<f32>,
+   c2v_consumer_buf = %c2v_import : i32,
+   v2c_consumer_buf = %c0_i32 : i32)
+
+// A5 (without GM slot buffer):
 pto.aic_initialize_pipe {dir_mask = 1, slot_size = 1024}
   (c2v_consumer_buf = %c2v_import : i32,
    v2c_consumer_buf = %c0_i32 : i32)
@@ -6820,7 +6827,7 @@ pto.aic_initialize_pipe {dir_mask = 1, slot_size = 1024}
 
 - `dir_mask`: communication direction encoding
 - `slot_size`: logical slot size in bytes
-- `gm_slot_buffer`: optional GM slot-buffer operand
+- `gm_slot_buffer`: optional GM pointer (`!pto.ptr<T>`), required on A2/A3, omitted on A5
 - `c2v_consumer_buf`: C2V consumer local base address
 - `v2c_consumer_buf`: V2C consumer local base address
 
@@ -6838,6 +6845,13 @@ pto.aic_initialize_pipe {dir_mask = 1, slot_size = 1024}
 **Syntax:**
 
 ```mlir
+// A2/A3 (with GM slot buffer):
+pto.aiv_initialize_pipe {dir_mask = 1, slot_size = 1024}
+  (gm_slot_buffer = %gm_buf : !pto.ptr<f32>,
+   c2v_consumer_buf = %c2v_local : i32,
+   v2c_consumer_buf = %c0_i32 : i32)
+
+// A5 (without GM slot buffer):
 pto.aiv_initialize_pipe {dir_mask = 1, slot_size = 1024}
   (c2v_consumer_buf = %c2v_local : i32,
    v2c_consumer_buf = %c0_i32 : i32)

--- a/docs/designs/ptoas-tpush-tpop-design.md
+++ b/docs/designs/ptoas-tpush-tpop-design.md
@@ -61,7 +61,7 @@ pto.aic_initialize_pipe(
 |---|---|---|
 | `DIR_MASK` | 编译期整数常量 | `1`、`2` 或 `3` |
 | `SLOT_SIZE` | 编译期整数常量 | 单 slot 字节数，定义为切分前完整 tile 字节数 |
-| `GM_SLOT_BUFFER` | GM 地址或空值 | A2/A3 路径使用，A5 路径为空 |
+| `GM_SLOT_BUFFER` | `!pto.ptr<T>` 或空值 | A2/A3 路径使用的 GM 指针，A5 路径为空 |
 | `C2V_CONSUMER_BUF` | `i32` | C2V 方向 consumer 的 local slot buffer 基址 |
 | `V2C_CONSUMER_BUF` | `i32` | V2C 方向 consumer 的 local slot buffer 基址 |
 

--- a/include/PTO/IR/PTOOps.td
+++ b/include/PTO/IR/PTOOps.td
@@ -1184,7 +1184,7 @@ def AicInitializePipeOp : PTO_Op<"aic_initialize_pipe"> {
   let arguments = (ins
       I8Attr:$dir_mask,
       I32Attr:$slot_size,
-      Optional<I32>:$gm_slot_buffer,
+      Optional<PtrType>:$gm_slot_buffer,
       I32:$c2v_consumer_buf,
       I32:$v2c_consumer_buf
   );
@@ -1210,7 +1210,7 @@ def AivInitializePipeOp : PTO_Op<"aiv_initialize_pipe"> {
   let arguments = (ins
       I8Attr:$dir_mask,
       I32Attr:$slot_size,
-      Optional<I32>:$gm_slot_buffer,
+      Optional<PtrType>:$gm_slot_buffer,
       I32:$c2v_consumer_buf,
       I32:$v2c_consumer_buf
   );

--- a/test/basic/tpush_tpop_frontend_lowering_a3.pto
+++ b/test/basic/tpush_tpop_frontend_lowering_a3.pto
@@ -2,7 +2,7 @@
 // RUN: ptoas --pto-arch=a3 --enable-insert-sync %s 2>&1 | FileCheck %s --check-prefix=SYNC-A3
 
 module {
-  func.func @cube_kernel(%gm_slot_buffer: i32)
+  func.func @cube_kernel(%gm_slot_buffer: !pto.ptr<f32>)
       attributes {pto.kernel_kind = #pto.kernel_kind<cube>} {
     %v2c_local = pto.reserve_buffer {
       name = "v2c_fifo",
@@ -15,7 +15,7 @@ module {
       peer_func = @vector_kernel
     } -> i32
     pto.aic_initialize_pipe {dir_mask = 3, slot_size = 1024}
-      (gm_slot_buffer = %gm_slot_buffer : i32,
+      (gm_slot_buffer = %gm_slot_buffer : !pto.ptr<f32>,
        c2v_consumer_buf = %c2v_import : i32,
        v2c_consumer_buf = %v2c_local : i32)
 
@@ -30,7 +30,7 @@ module {
     return
   }
 
-  func.func @vector_kernel(%gm_slot_buffer: i32)
+  func.func @vector_kernel(%gm_slot_buffer: !pto.ptr<f32>)
       attributes {pto.kernel_kind = #pto.kernel_kind<vector>} {
     %c2v_local = pto.reserve_buffer {
       name = "c2v_fifo",
@@ -43,7 +43,7 @@ module {
       peer_func = @cube_kernel
     } -> i32
     pto.aiv_initialize_pipe {dir_mask = 3, slot_size = 1024}
-      (gm_slot_buffer = %gm_slot_buffer : i32,
+      (gm_slot_buffer = %gm_slot_buffer : !pto.ptr<f32>,
        c2v_consumer_buf = %c2v_local : i32,
        v2c_consumer_buf = %v2c_import : i32)
 
@@ -60,7 +60,7 @@ module {
 
 }
 
-// A3-LABEL: AICORE void cube_kernel(
+// A3-LABEL: AICORE void cube_kernel(__gm__ float*
 // A3: auto {{v[0-9]+}} = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(
 // A3: TPUSH<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>
 // A3: Tile<TileType::Mat, float, 16, 16, BLayout::ColMajor, 16, 16, SLayout::RowMajor, 512, PadValue::Null> {{v[0-9]+}};
@@ -69,7 +69,7 @@ module {
 // A3: TMOV(
 // A3: TFREE<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, TileSplitAxis::TILE_NO_SPLIT>(
 
-// A3-LABEL: AICORE void vector_kernel(
+// A3-LABEL: AICORE void vector_kernel(__gm__ float*
 // A3: auto {{v[0-9]+}} = TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>(
 // A3: Tile<TileType::Vec, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::NoneBox, 512, PadValue::Null> {{v[0-9]+}};
 // A3: TPUSH<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, Tile<TileType::Vec, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::NoneBox, 512, PadValue::Null>, TileSplitAxis::TILE_NO_SPLIT>(
@@ -78,14 +78,14 @@ module {
 // A3: TNEG(
 // A3: TFREE<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, TileSplitAxis::TILE_NO_SPLIT>(
 
-// SYNC-A3-LABEL: AICORE void cube_kernel(
+// SYNC-A3-LABEL: AICORE void cube_kernel(__gm__ float*
 // SYNC-A3: TPOP<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, Tile<TileType::Mat, float, 16, 16, BLayout::ColMajor, 16, 16, SLayout::RowMajor, 512, PadValue::Null>, TileSplitAxis::TILE_NO_SPLIT>(
 // SYNC-A3: set_flag(PIPE_MTE2, PIPE_MTE1, EVENT_ID0);
 // SYNC-A3: Tile<TileType::Left, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::RowMajor, 512, PadValue::Null>
 // SYNC-A3: wait_flag(PIPE_MTE2, PIPE_MTE1, EVENT_ID0);
 // SYNC-A3: TMOV(
 
-// SYNC-A3-LABEL: AICORE void vector_kernel(
+// SYNC-A3-LABEL: AICORE void vector_kernel(__gm__ float*
 // SYNC-A3: TPOP<TPipe<0, Direction::DIR_BOTH, 1024, 4, 4>, Tile<TileType::Vec, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::NoneBox, 512, PadValue::Null>, TileSplitAxis::TILE_NO_SPLIT>(
 // SYNC-A3: set_flag(PIPE_MTE2, PIPE_V, EVENT_ID0);
 // SYNC-A3: Tile<TileType::Vec, float, 16, 16, BLayout::RowMajor, 16, 16, SLayout::NoneBox, 512, PadValue::Null>

